src/coreml.cs

```diff
@@ -0,0 +1,273 @@
+		MLFeatureValue FromInt64 (long value);
+
+		[Static]
+		[Export ("featureValueWithDouble:")]
+		MLFeatureValue FromDouble (double value);
+
+		[Static]
+		[Export ("featureValueWithString:")]
+		MLFeatureValue FromString (string value);
+
+		[Static]
+		[Export ("featureValueWithMultiArray:")]
+		MLFeatureValue FromMultiArray (MLMultiArray value);
+
+		[Static]
+		[Export ("undefinedFeatureValueWithType:")]
+		MLFeatureValue GetUndefinedFeatureValue (MLFeatureType type);
+
+		[Static]
+		[Export ("featureValueWithDictionary:error:")]
+		[return: NullAllowed]
+		MLFeatureValue FromDictionary (NSDictionary<NSObject, NSNumber> value, out NSError error);
+
+		[Export ("isEqualToFeatureValue:")]
+		bool IsEqualTo (MLFeatureValue value);
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLModel {
+
+		[Export ("modelDescription")]
+		MLModelDescription ModelDescription { get; }
+
+		[Static]
+		[Export ("modelWithContentsOfURL:error:")]
+		[return: NullAllowed]
+		MLModel FromUrl (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("predictionFromFeatures:error:")]
+		[return: NullAllowed]
+		IMLFeatureProvider GetPrediction (IMLFeatureProvider input, out NSError error);
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLModelDescription {
+
+		[Export ("inputDescriptionsByName")]
+		NSDictionary<NSString, MLFeatureDescription> InputDescriptionsByName { get; }
+
+		[Export ("outputDescriptionsByName")]
+		NSDictionary<NSString, MLFeatureDescription> OutputDescriptionsByName { get; }
+
+		[NullAllowed, Export ("predictedFeatureName")]
+		string PredictedFeatureName { get; }
+
+		[NullAllowed, Export ("predictedProbabilitiesName")]
+		string PredictedProbabilitiesName { get; }
+
+		[Export ("metadata")]
+		[Internal]
+		NSDictionary _Metadata { get; }
+
+		[Wrap ("_Metadata")]
+		MLModelMetadata Metadata { get; }
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Static]
+	interface MLModelMetadataKeys {
+
+		[Field ("MLModelDescriptionKey")]
+		NSString DescriptionKey { get; }
+
+		[Field ("MLModelVersionStringKey")]
+		NSString VersionStringKey { get; }
+
+		[Field ("MLModelAuthorKey")]
+		NSString AuthorKey { get; }
+
+		[Field ("MLModelLicenseKey")]
+		NSString LicenseKey { get; }
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[StrongDictionary ("MLModelMetadataKeys")]
+	interface MLModelMetadata {
+		string Description { get; }
+		string VersionString { get; }
+		string Author { get; }
+		string License { get; }
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject))]
+	interface MLMultiArray {
+
+		[Export ("dataPointer")]
+		IntPtr DataPointer { get; }
+
+		[Export ("dataType")]
+		MLMultiArrayDataType DataType { get; }
+
+		[Export ("shape")]
```

Comment:
Unfortunately, docs are not clear on the type we should use here nor is swift
`var shape: [NSNumber]`